devicetypes/smartthings/zigbee-multi-switch.src/zigbee-multi-switch.groovy

```diff
@@ -165,14 +168,45 @@ def parse(String description) {
 	}
 }
 
-private void createChildDevices() {    
-	if (!childDevices) {
-		def x = getChildCount()
-		for (i in 2..x) {
-			addChildDevice("Child Switch Health", "${device.deviceNetworkId}:0${i}", device.hubId,
-				[completedSetup: true, label: "${device.displayName[0..-2]}${i}", isComponent: false])
-		}
-	}
+private void createChildDevices() { 
+    if (isNodOn) {
+        if (!state.isCreateChildDone || isAutoCreateChildDevice != false) {
+            def endpointCount = 2
+            def endpointId = device.getDataValue("endpointId")
+            def endpointInt = zigbee.convertHexToInt(endpointId)
+            def deviceLabel = "${device.displayName[0..-2]}"
+            deviceLabel.minus("1")
+
+            for (i in 1..endpointCount - 1) {
+                def endpointHexString = zigbee.convertToHexString(endpointInt + i, 2).toUpperCase()
+                createChildDevice("$deviceLabel${i + 1}", endpointHexString)
+            }
+
+
+            state.isCreateChildDone = true
+        }
+    }else{   
+        if (!childDevices) {
+            def x = getChildCount()
+            for (i in 2..x) {
+                addChildDevice("Child Switch Health", "${device.deviceNetworkId}:0${i}", device.hubId,
+                    [completedSetup: true, label: "${device.displayName[0..-2]}${i}", isComponent: false])
+            }
+        }
+    }
+}
+
+private void createChildDevice(String deviceLabel, String endpointHexString) {
+    def childDevice = childDevices.find {
+        it.deviceNetworkId == "$device.deviceNetworkId:$endpointHexString"
+    }
+    if (!childDevice) {
+        log.debug("===========Need to createChildDevice: $device.deviceNetworkId:$endpointHexString")
+        addChildDevice("smartthings", "Child Switch", "$device.deviceNetworkId:$endpointHexString", device.hubId,
+                       [completedSetup: true, label: deviceLabel, isComponent: false])
+    } else {
+        log.debug("createChildDevice: SKIP - $device.deviceNetworkId:${endpointHexString}")
+    }
```

Comment:
You're seriously over-complicating this. Even if you wouldn't introduce your own code for creating child device, its label should still be `NodOn Light2`. If not, elaborate more about what's wrong, there definitely should be a more elegant way to deal with this.